### ISSUE TITLE
ALIAS: Option to use ALIAS TTL as TTL or as "minimum" TTL

### ISSUE THREAD

**Issue by DarkDeviL** [PowerDNS/pdns#6874]
- Program: Authoritative 
 - Issue type: Feature request 

### Short description

It would be nice with a option to use the ALIAS TTL as the TTL (or minimum TTL) for the provided A/AAAA record.

It seems like there has previously been an attempt, but without much discussion / follow ups regarding the TTL on the mailing list thread [Please test: ALIAS/ANAME apex record in PowerDNS @ Sun Sep 21 10:54:07 UTC 2014](https://mailman.powerdns.com/pipermail/pdns-users/2014-September/010921.html) from @ahupowerdns:

> Finally, for TTL, we currently use what the resolver gave us. But perhaps we
> could use the TTL of the ALIAS record instead, or as a maximum? Or minimum?

### Usecase

Providing longer DNS caches instead of a million DNS requests (sometimes imposed due to impractical policies decided by third parties).

### Description

There are some CDN providers out there, giving out a 1 second TTL on their CDN RR, e.g. it is not unlikely to see something like:

```
example.net.    300    IN      ALIAS   www.example.net.cdn.example.org.
www.example.net.cdn.example.org.    3600    IN      CNAME   dk.eu.l.example.org.
dk.eu.l.example.org. 1 IN      A       192.0.2.245
dk.eu.l.example.org. 1 IN      A       198.51.100.245
dk.eu.l.example.org. 1 IN      A       203.0.113.245
```
Currently, ALIAS seems (rightfully, according to ANAME specifications), to be using the minimum TTL specified in the whole DNS chain, setting the TTL of "example.net" to 1 in this demonstrated example.

Sometimes you're even routed over and over to the same individual IP address (or same two), even though those providers do those low TTL's.

It could be done as a setting, e.g. (default= no):

```
alias-use-ttl-as-minimum-ttl=yes
```

where if =yes, then the TTL would be 300, not 1.

In short words, a possible way to override and/or adjust the TTL.
